penguin/llm/runtime.py

```
@@ -287,40 +328,93 @@ async def execute_pending_tool_call(
+        raw_args = tool_call.arguments
+        raw_args_text = raw_args if isinstance(raw_args, str) else str(raw_args)
+        raw_args_by_id[tool_call.id] = raw_args_text
+        try:
+            import json as _json
+
+            parsed_args = (
+                _json.loads(raw_args_text)
+                if isinstance(raw_args_text, str) and raw_args_text.strip()
+                else {}
+            )
+        except Exception:
```

Ruff (0.15.11)

[warning] Do not catch blind exception: `Exception`

(BLE001)